frontend/src/features/admin-form/responses/ResponsesPage/storage/UnlockedResponses/DownloadWithAttachmentModal/DownloadWithAttachmentModal.tsx

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useState } from 'react'
+import {
+  Modal,
+  ModalContent,
+  ModalOverlay,
+  Text,
+  UseDisclosureReturn,
+} from '@chakra-ui/react'
+
+import { XMotionBox } from '~templates/MotionBox'
+
+import { ProgressModalContent } from '../ProgressModal'
+
+import { ConfirmationScreen } from './ConfirmationScreen'
+
+export interface DownloadWithAttachmentModalProps
+  extends Pick<UseDisclosureReturn, 'onClose' | 'isOpen'> {
+  onDownload: () => void
+  onCancel: () => void
+  isDownloading: boolean
+  responsesCount: number
+  downloadPercentage: number
+}
+
+export enum DownloadWithAttachmentFlowStates {
+  Confirmation = 'confirmation',
+  Progress = 'progress',
+  Complete = 'complete',
```

Comment:
When would be using the `DownloadWithAttachmentFlowStates.Complete` state?

Reply:
coming in next pr, for download completion summary.